visualization/app/codeCharta/ui/attributeSideBar/ccNodePath/ccNodePath.component.spec.ts

```diff
@@ -0,0 +1,86 @@
+import angular from "angular"
+import "angular-mocks"
+
+import "../attributeSideBar.module"
+import { ccNodePathComponent } from "./ccNodePath.component"
+
+describe("CcNodePathComponent", () => {
+	let $compile
+	let $rootScope
+
+	beforeEach(() => {
+		angular.mock.module("app.codeCharta.ui.attributeSideBar")
+
+		angular.mock.inject(function (_$compile_, _$rootScope_) {
+			$compile = _$compile_
+			$rootScope = _$rootScope_
+		})
+	})
+
+	afterEach(() => {
+		jest.clearAllMocks()
+	})
+
+	it("should update its `_viewModel` when a `building-selected`-event is fired", () => {
+		const component = $compile("<cc-node-path-component></cc-node-path-component>")($rootScope)
+		$rootScope.$emit("building-selected", {
+			node: {
+				isLeaf: false,
+				path: "some/file.ts",
+				attributes: { unary: 1 }
+			}
+		})
+		$rootScope.$digest()
+
+		const viewModel = component.controller("ccNodePathComponent")._viewModel
+		expect(viewModel.node.isLeaf).toBe(false)
+		expect(viewModel.node.path).toBe("some/file.ts")
+		expect(viewModel.packageFileCount).toBe(1)
+		expect(viewModel.fileCountDescription).toBe("1 file")
+	})
+
+	it("should display path of selected building", () => {
+		const component = $compile("<cc-node-path-component></cc-node-path-component>")($rootScope)
+		$rootScope.$emit("building-selected", {
+			node: { isLeaf: true, path: "some/file.ts" }
+		})
+		$rootScope.$digest()
+
+		expect(component.text()).toContain("some/file.ts")
+	})
+
+	it("should display no file count information if selected building is a leaf / i.e. a file", () => {
+		const component = $compile("<cc-node-path-component></cc-node-path-component>")($rootScope)
+		$rootScope.$emit("building-selected", {
+			node: { isLeaf: true, path: "some/file.ts" }
+		})
+		$rootScope.$digest()
+
+		expect(component[0].getElementsByClassName("cc-node-file-count").length).toBe(0)
+	})
```

Comment:
I think we can combine this two tests.

Reply:
From my point of view they test different pieces of logic and should therefore be kept separated. As they take less than 100ms I would keep them separated. Does that makes sense to you?